src-java/opentsdb-topology/opentsdb-storm-topology/src/main/java/org/openkilda/wfm/topology/opentsdb/service/FilterService.java

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.opentsdb.service;
+
+import org.openkilda.messaging.info.Datapoint;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Slf4j
+public class FilterService {
+    static final long MUTE_IF_NO_UPDATES_SECS = TimeUnit.MINUTES.toSeconds(10);
+    static final long MUTE_IF_NO_UPDATES_MILLIS = TimeUnit.SECONDS.toMillis(MUTE_IF_NO_UPDATES_SECS);
+    @VisibleForTesting
+    Map<DatapointKey, Datapoint> storage = new HashMap<>();
+    private DatapointCarrier carrier;
+
+
+    public FilterService(DatapointCarrier carrier) {
+        this.carrier = carrier;
+    }
+
+    /**
+     * Handle periodic events.
+     */
+    public void handlePeriodic() {
+        long now = System.currentTimeMillis();
+        storage.entrySet().removeIf(entry -> now - entry.getValue().getTime() > MUTE_IF_NO_UPDATES_MILLIS);
+
+        if (log.isTraceEnabled()) {
+            log.trace("storage after clean tuple: {}", storage.toString());
+        }
+    }
+
+    /**
+     * Handle new datapoint.
+     */
+    public void handleData(Datapoint datapoint) {
+        if (isUpdateRequired(datapoint)) {
+            addDatapoint(datapoint);
+
+            List<Object> stream = Stream.of(datapoint.getMetric(), datapoint.getTime(), datapoint.getValue(),
```

Comment:
Here you are mixing service and trasport (carrier) responsibilities i.e. service makes transport level encoding.